source/common/protobuf/utility.h

```diff
@@ -109,6 +116,42 @@ class MessageUtil {
   static void loadFromYaml(const std::string& yaml, Protobuf::Message& message);
   static void loadFromFile(const std::string& path, Protobuf::Message& message);
 
+  /**
+   * Validate protoc-gen-validate constraints on a given protobuf.
+   * @param message message to validate.
+   * @throw ProtoValidationException if the message does not satisfy its type constraints.
+   */
+  template <class MessageType> static void validate(const MessageType& message) {
+    std::string err;
+    if (!Validate(message, &err)) {
+      throw ProtoValidationException(err, message);
+    }
+  }
+
+  template <class MessageType>
+  static void loadFromFileAndValidate(const std::string& path, MessageType& message) {
+    loadFromFile(path, message);
+    try {
+      validate(message);
+    } catch (const EnvoyException&) {
+      message.Clear();
```

Comment:
what's the reasoning behind clearing here? We don't do it in the other functions where we validate. Just because we also loaded here so we own the whole thing? Maybe comment?

Reply:
Yeah, it had to do with some later code that depended on this being clear on failure in server init. TBH, that's an anti-pattern, I will fix that code instead.